pyomo/opt/results/solver.py

```diff
@@ -68,6 +68,33 @@
     )
 
 
+def check_optimal_termination(results, suppress_exception=False):
```

Comment:
With `raise` by default, this doesn't strike me as a `check` function. It's more of an `assert` function, in which case it is odd for it to be returning `True`/`False` values. I see what you're trying to accomplish, but wouldn't it make more sense to retain the original behavior (and not have to add a flag) by just using `assert optimal_termination(results)` when you need a one-liner?

Reply:
@qtothec makes a good point...